source/extensions/matching/http/cel_input/cel_input.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+#include "envoy/matcher/matcher.h"
+#include "envoy/server/factory_context.h"
+#include "envoy/type/matcher/v3/http_inputs.pb.h"
+#include "envoy/type/matcher/v3/http_inputs.pb.validate.h"
+
+#include "source/common/http/header_utility.h"
+#include "source/common/http/utility.h"
+#include "source/extensions/filters/common/expr/evaluator.h"
+#include "source/extensions/matching/input_matchers/cel_matcher/matcher.h"
+
+#include "xds/type/matcher/v3/http_inputs.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace Http {
+namespace CelInput {
+
+using ::Envoy::Http::RequestHeaderMapOptConstRef;
+using ::Envoy::Http::ResponseHeaderMapOptConstRef;
+using ::Envoy::Http::ResponseTrailerMapOptConstRef;
+
+using ::Envoy::Extensions::Matching::InputMatchers::CelMatcher::CelMatchData;
+
+class HttpCelDataInput : public Matcher::DataInput<Envoy::Http::HttpMatchingData> {
+public:
+  HttpCelDataInput() = default;
+  Matcher::DataInputGetResult get(const Envoy::Http::HttpMatchingData& data) const override {
+    RequestHeaderMapOptConstRef maybe_request_headers = data.requestHeaders();
+    ResponseHeaderMapOptConstRef maybe_response_headers = data.responseHeaders();
+    ResponseTrailerMapOptConstRef maybe_response_trailers = data.responseTrailers();
+
+    // Returns NotAvailable state when all of three below are empty.
+    if (!maybe_request_headers && !maybe_response_headers && !maybe_response_trailers) {
+      return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, absl::monostate()};
```

Comment:
I think it probably makes sense to return CEL activation with only StreamInfo. It's still useful. Does this path ever get hit at all in the prod code?

Reply:
This path will get hit as my test `CelMatcherNoRequestAttributes` shows.

Will it get hit in production? probably not. But I think it doesn't hurt to have it.